kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -79,71 +75,81 @@ public void initChannel(SocketChannel ch) {
         var bindingAddress = ch.parent().localAddress().getAddress().isAnyLocalAddress() ? Optional.<String> empty()
                 : Optional.of(ch.localAddress().getAddress().getHostAddress());
         if (tls) {
-            LOGGER.debug("Adding SSL/SNI handler");
-            pipeline.addLast(new SniHandler((sniHostname, promise) -> {
-                try {
-                    var stage = virtualClusterBindingResolver.resolve(Endpoint.createEndpoint(bindingAddress, targetPort, tls), sniHostname);
-                    // completes the netty promise when then resolution completes (success/otherwise).
-                    var unused = stage.handle((binding, t) -> {
-                        try {
+            initTlsChannel(ch, pipeline, bindingAddress, targetPort);
+        }
+        else {
+            initPlainChannel(ch, pipeline, bindingAddress, targetPort);
+        }
+    }
+
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private void initPlainChannel(SocketChannel ch, ChannelPipeline pipeline, Optional<String> bindingAddress, int targetPort) {
+        pipeline.addLast(new ChannelInboundHandlerAdapter() {
+            @Override
+            public void channelActive(ChannelHandlerContext ctx) {
+                virtualClusterBindingResolver.resolve(Endpoint.createEndpoint(bindingAddress, targetPort, tls), null)
+                        .handle((binding, t) -> {
                             if (t != null) {
-                                promise.setFailure(t);
+                                ctx.fireExceptionCaught(t);
```

Comment:
Why was this necessary (I'm not say it's wrong, just interested)

Reply:
I don't remember changing that. I'm more than a little suspicious the diff is lying(well getting confused) but I will dig further once I finish bashing at a unit test to generate some coverage.

Reply:
There is a reverse diff around line [129/120](https://github.com/kroxylicious/kroxylicious/pull/905/files/bd12dbbd5f7e26578cc76e4e5528d7b464e7e795#diff-90761a096b3920e7984e680dae477133069134d19d4b01297f42e528531e26c7L129)

Reply:
The diff I see in IntelliJ doesn't have this change. It shows two big blocks of change rather than small line by line changes.